src/main/mule/sources/anypoint/platform/apis/api-call-api-platform.xml

```diff
@@ -28,6 +29,61 @@ http://www.mulesoft.org/schema/mule/ee/core http://www.mulesoft.org/schema/mule/
 	"targetAdminSite" : p('anypoint.platform.apis.apiPlatform.clients.targetAdminSite')
 }]]]></http:query-params>
 		</http:request>
+		<ee:transform doc:name="Get number of API Clients Pages" doc:id="22c75968-ad73-48dc-a94e-3f41253842e0" >
+			<ee:message >
+			</ee:message>
+			<ee:variables >
+				<ee:set-variable variableName="apiClientsPages" ><![CDATA[%dw 2.0
+output application/java
+var pageLimit = (Mule::p('anypoint.platform.apis.apiPlatform.clients.limit') as Number default 100)
+---
+((1 to ceil((vars.apiClientsResponsePayload."total" default 0)/pageLimit)) - 1 default [])]]></ee:set-variable>
+			</ee:variables>
+		</ee:transform>
+		<logger level="INFO" doc:name="Logger" doc:id="4fe0e979-f9be-46be-b475-97cee57a21e1" message="limit #[Mule::p('anypoint.platform.apis.apiPlatform.clients.limit')] offset p('anypoint.platform.apis.apiPlatform.clients.offset')"/>
+		<foreach doc:name="For Each" doc:id="0268328b-51dc-4934-83e6-174f243d52cf" collection="#[vars.apiClientsPages]">
```

Comment:
@roltra Shouldn't we provide a choice in case the first call returned all clients? If not we are making a call twice if pages=1

Reply:
The DW expression should return an empty collection (hence no foreach iteration) if the total number of apps is equal or smaller than the limit, unless I missed something in my code, although a choice element could make that more readable for sure

Reply:
As Rafael explained, the "- 1" in the previous DW expression removes the first page, so the first requests is already done and API is not called twice

Reply:
I have only one requirement, could you improve this log? Mixed with other requests, I wouldn't understand what it means. Also, I don't know why a DW Expression is declared first, but not later (The second one shouldn't work). You can just reference a configuration property using ${}